miden-lib/asm/sat/layout.masm

```diff
@@ -389,6 +389,16 @@ export.set_acct_nonce
     drop movup.3 push.ACCT_ID_AND_NONCE_PTR mem_storew dropw
 end
 
+#! Sets the code root of the account.
+#!
+#! Stack: [CODE_ROOT]
+#! Output: []
+#!
+#! - CODE_ROOT is the code root to be set.
+export.set_acct_code_root
+    push.ACCT_CODE_ROOT_PTR mem_storew dropw
+end
```

Comment:
This would update the account code root immediately, right? If so, I wonder if this is the right approach. The reason is that for access checks (#79) we'd need to check if a procedure exists against the code root. So, changing it immediately would mean that the user would be able to do a "hot swap" of the code, including any tx authentication procedures. This may or may not be a problem - but I wonder if we should be conservative here.

So, I guess we need to decide:
1. Do we update the code root immediately?
2. Do we delay the update until the epilogue? (i.e., cache the new code root somewhere, and copy it over in the epilogue).

My preliminary thinking is that option 2 is better as it is simpler to think through and much less complex (even from the advice provider standpoint).

Reply:
I think either approach is valid and (most likely) practical. I recognise that there may be some additional complexity around advice provider interaction with immediate code root update. I don't have a strong preference either way.  I'll push on with option 2 for now.  I think the decision should be made based on UX.

 One question on this topic is whether we allow the `set_code` procedure to be called multiple times in a transaction and what the expected behaviour should be in this scenario?